monitoring-dashboard/components/org.wso2.ei.dashboard.core/src/main/java/org/wso2/ei/dashboard/core/rest/delegates/nodes/NodesDelegate.java

```diff
@@ -52,4 +55,41 @@ public NodeList getNodes(String groupId) {
         }
         return nodeList;
     }
+
+    public NodeList getPaginatedNodesList(String groupId, int lowerLimit, int upperLimit) {  
+        NodeList nodeList = getNodes(groupId);
+        NodeList resultList = new NodeList();
+        try {
+            if (nodeList.size() < upperLimit) {
+                upperLimit = nodeList.size();
+            }
+            if (upperLimit < lowerLimit) {
+                lowerLimit = upperLimit;
+            }
+            List<NodeListInner> paginatedList = nodeList.subList(lowerLimit, upperLimit);
+            for (NodeListInner node : paginatedList) {
+                resultList.add(node);
+            }
+            return resultList;
+        } catch (IndexOutOfBoundsException e) {
+            throw new IndexOutOfBoundsException("Index values are out of bound");
+        } catch (IllegalArgumentException e) {
+            throw new IllegalArgumentException("Illegal arguments for index values");
```

Comment:
Shall we pass the caught exception too. It would be helpful to troubleshoot the issue.